timelock-corruption-detection/src/main/java/com/palantir/history/models/LearnerAndAcceptorRecords.java

```diff
@@ -31,4 +33,34 @@
 
     @Value.Parameter
     Map<Long, PaxosAcceptorState> acceptorRecords();
+
+    default Optional<PaxosValue> getLearnedValueAtSeqIfExists(long seq) {
+        return Optional.ofNullable(learnerRecords().getOrDefault(seq, null));
+    }
+
+    default Optional<PaxosAcceptorState> getAcceptedValueAtSeqIfExists(long seq) {
+        return Optional.ofNullable(acceptorRecords().getOrDefault(seq, null));
+    }
+
+    default long getMinSequence() {
+        long minSeq = Long.MAX_VALUE;
+        if (!learnerRecords().isEmpty()) {
+            minSeq = Math.min(minSeq, Collections.min(learnerRecords().keySet()));
+        }
+        if (!acceptorRecords().isEmpty()) {
+            minSeq = Math.min(minSeq, Collections.min(acceptorRecords().keySet()));
+        }
+        return minSeq;
+    }
+
+    default long getMaxSequence() {
+        long maxSeq = Long.MIN_VALUE;
+        if (!learnerRecords().isEmpty()) {
+            maxSeq = Math.max(maxSeq, Collections.max(learnerRecords().keySet()));
+        }
+        if (!acceptorRecords().isEmpty()) {
+            maxSeq = Math.max(maxSeq, Collections.max(acceptorRecords().keySet()));
+        }
+        return maxSeq;
```

Comment:
as above, I don't know if we want to cache the value for recomputations, and I think streams give an easier pipeline.